lib/travis/api/app.rb

```diff
@@ -81,11 +82,7 @@ def self.new(options = {})
     end
 
     def self.deploy_sha
-      @deploy_sha ||= File.exist?(deploy_sha_path) ? File.read(deploy_sha_path)[0..7] : 'deploy-sha'
-    end
-
-    def self.deploy_sha_path
-      File.expand_path('../../../../.deploy-sha', __FILE__)
+      @deploy_sha ||= ENV['HEROKU_SLUG_COMMIT'] || SecureRandom.hex(5)
```

Comment:
It's so that the key changes with every deploy for [cache invalidation](https://github.com/travis-ci/travis-api/blob/8a050a468c77f2ad8c5faa99864c0bd7d61b08a0/lib/travis/api/app/responders/service.rb#L39-L44).